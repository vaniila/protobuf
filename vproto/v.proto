syntax = "proto3";

package vproto;

import "github.com/vaniila/protobuf/gogoproto/gogo.proto";

enum ErrorType {
    option (gogoproto.goproto_enum_prefix) = false;
    Unknown = 0 [(gogoproto.enumvalue_customname) = "Unknown"];
    System = 1 [(gogoproto.enumvalue_customname) = "System"];
    Client = 2 [(gogoproto.enumvalue_customname) = "Client"];
}

message Error {
    string cause = 1;
    ErrorType type = 2;
}

message ExtendedRequest {
    map<string, string> request_metadata = 1;
    bool option_error_if_null = 10;
    bool option_error_if_zero = 11;
}

message ExtendedResponse {
    Error error = 10;
}

