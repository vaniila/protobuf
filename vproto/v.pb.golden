// Code generated by protoc-gen-gogo. DO NOT EDIT.
// source: v.proto

/*
Package vproto is a generated protocol buffer package.

It is generated from these files:
	v.proto

It has these top-level messages:
	Error
	ExtendedRequest
	ExtendedResponse
*/
package vproto

import proto "github.com/vaniila/protobuf/proto"
import fmt "fmt"
import math "math"
import _ "github.com/vaniila/protobuf/gogoproto"

// Reference imports to suppress errors if they are not otherwise used.
var _ = proto.Marshal
var _ = fmt.Errorf
var _ = math.Inf

// This is a compile-time assertion to ensure that this generated file
// is compatible with the proto package it is being compiled against.
// A compilation error at this line likely means your copy of the
// proto package needs to be updated.
const _ = proto.GoGoProtoPackageIsVersion2 // please upgrade the proto package

type ErrorType int32

const (
	Unknown ErrorType = 0
	System  ErrorType = 1
	Client  ErrorType = 2
)

var ErrorType_name = map[int32]string{
	0: "Unknown",
	1: "System",
	2: "Client",
}
var ErrorType_value = map[string]int32{
	"Unknown": 0,
	"System":  1,
	"Client":  2,
}

func (x ErrorType) String() string {
	return proto.EnumName(ErrorType_name, int32(x))
}
func (ErrorType) EnumDescriptor() ([]byte, []int) { return fileDescriptorV, []int{0} }

type Error struct {
	Cause string    `protobuf:"bytes,1,opt,name=cause,proto3" json:"cause,omitempty"`
	Type  ErrorType `protobuf:"varint,2,opt,name=type,proto3,enum=vproto.ErrorType" json:"type,omitempty"`
}

func (m *Error) Reset()                    { *m = Error{} }
func (m *Error) String() string            { return proto.CompactTextString(m) }
func (*Error) ProtoMessage()               {}
func (*Error) Descriptor() ([]byte, []int) { return fileDescriptorV, []int{0} }

func (m *Error) GetCause() string {
	if m != nil {
		return m.Cause
	}
	return ""
}

func (m *Error) GetType() ErrorType {
	if m != nil {
		return m.Type
	}
	return Unknown
}

type ExtendedRequest struct {
	RequestMetadata   map[string]string `protobuf:"bytes,1,rep,name=request_metadata,json=requestMetadata" json:"request_metadata,omitempty" protobuf_key:"bytes,1,opt,name=key,proto3" protobuf_val:"bytes,2,opt,name=value,proto3"`
	OptionErrorIfNull bool              `protobuf:"varint,10,opt,name=option_error_if_null,json=optionErrorIfNull,proto3" json:"option_error_if_null,omitempty"`
	OptionErrorIfZero bool              `protobuf:"varint,11,opt,name=option_error_if_zero,json=optionErrorIfZero,proto3" json:"option_error_if_zero,omitempty"`
}

func (m *ExtendedRequest) Reset()                    { *m = ExtendedRequest{} }
func (m *ExtendedRequest) String() string            { return proto.CompactTextString(m) }
func (*ExtendedRequest) ProtoMessage()               {}
func (*ExtendedRequest) Descriptor() ([]byte, []int) { return fileDescriptorV, []int{1} }

func (m *ExtendedRequest) GetRequestMetadata() map[string]string {
	if m != nil {
		return m.RequestMetadata
	}
	return nil
}

func (m *ExtendedRequest) GetOptionErrorIfNull() bool {
	if m != nil {
		return m.OptionErrorIfNull
	}
	return false
}

func (m *ExtendedRequest) GetOptionErrorIfZero() bool {
	if m != nil {
		return m.OptionErrorIfZero
	}
	return false
}

type ExtendedResponse struct {
	Error *Error `protobuf:"bytes,10,opt,name=error" json:"error,omitempty"`
}

func (m *ExtendedResponse) Reset()                    { *m = ExtendedResponse{} }
func (m *ExtendedResponse) String() string            { return proto.CompactTextString(m) }
func (*ExtendedResponse) ProtoMessage()               {}
func (*ExtendedResponse) Descriptor() ([]byte, []int) { return fileDescriptorV, []int{2} }

func (m *ExtendedResponse) GetError() *Error {
	if m != nil {
		return m.Error
	}
	return nil
}

func init() {
	proto.RegisterType((*Error)(nil), "vproto.Error")
	proto.RegisterType((*ExtendedRequest)(nil), "vproto.ExtendedRequest")
	proto.RegisterType((*ExtendedResponse)(nil), "vproto.ExtendedResponse")
	proto.RegisterEnum("vproto.ErrorType", ErrorType_name, ErrorType_value)
}

func init() { proto.RegisterFile("v.proto", fileDescriptorV) }

var fileDescriptorV = []byte{
	// 370 bytes of a gzipped FileDescriptorProto
	0x1f, 0x8b, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00, 0x02, 0xff, 0x6c, 0x50, 0x4d, 0x6b, 0xdb, 0x40,
	0x14, 0xb4, 0xe4, 0xaf, 0xfa, 0x89, 0xd6, 0xf2, 0x62, 0x8a, 0xd0, 0xa1, 0x08, 0x97, 0x82, 0x28,
	0x45, 0x2a, 0xee, 0xa1, 0xa5, 0xc7, 0x24, 0x3e, 0xe4, 0x90, 0x1c, 0x94, 0x84, 0x40, 0x72, 0x10,
	0xb2, 0xfd, 0xec, 0x08, 0xcb, 0xbb, 0xca, 0x6a, 0xa5, 0x44, 0xf9, 0x05, 0xc1, 0xa7, 0xfc, 0x01,
	0x9f, 0xf2, 0x47, 0x83, 0x76, 0x15, 0x07, 0x83, 0x4f, 0xef, 0xcd, 0x8c, 0x46, 0x6f, 0x76, 0xa0,
	0x5b, 0x78, 0x29, 0x67, 0x82, 0x91, 0x4e, 0x21, 0xa7, 0xfd, 0x7b, 0x19, 0x8b, 0xbb, 0x7c, 0xea,
	0xcd, 0xd8, 0xda, 0x2f, 0x22, 0x1a, 0xc7, 0x49, 0xe4, 0x4b, 0x65, 0x9a, 0x2f, 0xfc, 0x25, 0x5b,
	0x32, 0x09, 0xe4, 0xa6, 0x9c, 0xa3, 0x13, 0x68, 0x4f, 0x38, 0x67, 0x9c, 0x0c, 0xa1, 0x3d, 0x8b,
	0xf2, 0x0c, 0x2d, 0xcd, 0xd1, 0xdc, 0x5e, 0xa0, 0x00, 0xf9, 0x01, 0x2d, 0x51, 0xa6, 0x68, 0xe9,
	0x8e, 0xe6, 0x7e, 0x19, 0x0f, 0x3c, 0x75, 0xc7, 0x93, 0x96, 0xcb, 0x32, 0xc5, 0x40, 0xca, 0xa3,
	0x17, 0x1d, 0xfa, 0x93, 0x47, 0x81, 0x74, 0x8e, 0xf3, 0x00, 0xef, 0x73, 0xcc, 0x04, 0xb9, 0x06,
	0x93, 0xab, 0x35, 0x5c, 0xa3, 0x88, 0xe6, 0x91, 0x88, 0x2c, 0xcd, 0x69, 0xba, 0xc6, 0xf8, 0xd7,
	0xee, 0x37, 0xfb, 0x16, 0xaf, 0x9e, 0x67, 0xf5, 0xe7, 0x13, 0x2a, 0x78, 0x19, 0xf4, 0xf9, 0x3e,
	0x4b, 0x7c, 0x18, 0xb2, 0x54, 0xc4, 0x8c, 0x86, 0x58, 0xc5, 0x08, 0xe3, 0x45, 0x48, 0xf3, 0x24,
	0xb1, 0xc0, 0xd1, 0xdc, 0x4f, 0xc1, 0x40, 0x69, 0x32, 0xe1, 0xe9, 0xe2, 0x3c, 0x4f, 0x92, 0x43,
	0x86, 0x27, 0xe4, 0xcc, 0x32, 0x0e, 0x18, 0x6e, 0x90, 0x33, 0xfb, 0x08, 0x86, 0x87, 0xa2, 0x10,
	0x13, 0x9a, 0x2b, 0x2c, 0xeb, 0x86, 0xaa, 0xb5, 0x6a, 0xad, 0x88, 0x92, 0x5c, 0x15, 0xd4, 0x0b,
	0x14, 0xf8, 0xaf, 0xff, 0xd3, 0x46, 0x7f, 0xc1, 0xfc, 0x78, 0x5e, 0x96, 0x32, 0x9a, 0x21, 0xf9,
	0x0e, 0x6d, 0x99, 0x40, 0x46, 0x35, 0xc6, 0x9f, 0xf7, 0xea, 0x0c, 0x94, 0xf6, 0xf3, 0x16, 0x7a,
	0xbb, 0x7a, 0x89, 0x05, 0xdd, 0x2b, 0xba, 0xa2, 0xec, 0x81, 0x9a, 0x0d, 0xdb, 0xd8, 0x6c, 0x9d,
	0x77, 0x48, 0xbe, 0x42, 0xe7, 0xa2, 0xcc, 0x04, 0xae, 0x4d, 0xcd, 0x86, 0xcd, 0xd6, 0xa9, 0x51,
	0xc5, 0x1f, 0x27, 0x31, 0x52, 0x61, 0xea, 0x8a, 0x57, 0xc8, 0x6e, 0x3d, 0xbf, 0x7e, 0x6b, 0x4c,
	0x3b, 0xf2, 0xe0, 0x9f, 0xb7, 0x00, 0x00, 0x00, 0xff, 0xff, 0x43, 0x1f, 0xc3, 0x9f, 0x3a, 0x02,
	0x00, 0x00,
}
